ygen/gogen.go

```diff
@@ -730,6 +750,19 @@ func (t *{{ .Receiver }}) Get{{ .Name }}() {{ .Type }} {
 	}
 	return {{ if .IsPtr -}} * {{- end -}} t.{{ .Name }}
 }
+`
+
+	// goLeafSetterTemplate defines a template for a function that, for a
+	// particular leaf, generates a setter method.
+	goLeafSetterTemplate = `
+// Set{{ .Name }} sets the value of the leaf {{ .Name }} from the {{ .Receiver }}
+// struct. 
+func (t *{{ .Receiver }}) Set{{ .Name }}(val string) {
+	if t == nil || val == "" {
+		return
+	}
+	t.{{ .Name }} = ygot.String(val)
```

Comment:
Same comment as above re: whether you can use `string`.

This likely highlights a test coverage gap -- we should add tests to `schema_tests` which give an integration test against a YANG model that has >1 different field type.